### PR TITLE
fix: parse object.id in performUndo() and add tests

### PR DESCRIPTION
**Pull request by magush27**
Fixed issue https://github.com/hyphacoop/distributed-press-organizing/issues/150

### CODE REVIEW

File: src/server/apsystem.test.ts [hunk 1]
```diff
@@ -215,6 +215,65 @@ test('ActivityPubSystem - List replies', async t => {
   t.deepEqual(collection.items, [activity.object])
 })
 
+test('ActivityPubSystem - Undo activity', async t => {
+  const store = newStore()
+  const hookSystem = new HookSystem(store, mockFetch)
+  const aps = new ActivityPubSystem('http://localhost', store, mockModCheck, hookSystem, mockServer.log)
```
Comment (akhileshthite) [inline src/server/apsystem.test.ts:221]: Maybe add a helper function to set up the ActivityPubSystem for these tests. Also, make sure to update the `replies` test.

```javascript
function setupActivityPubSystem () {
  const store = newStore()
  const hookSystem = new HookSystem(store, mockFetch)
  const aps = new ActivityPubSystem('http://localhost', store, mockModCheck, hookSystem, mockServer.log)
  return { store, aps }
}
```

```suggestion
  const { store, aps } = setupActivityPubSystem()
```

File: src/server/apsystem.test.ts [hunk 2]
```diff
@@ -215,6 +215,65 @@ test('ActivityPubSystem - List replies', async t => {
...
+      id: activity.id,
+      type: activity.type
+    },
+    id: 'https://example.com/undo1'
+  }
+
+  await store.forActor(actorMention).inbox.add(activity)
+  await aps.approveActivity(actorMention, activity.id as string)
+
+  // Log to verify the activity is in the store
```
Comment (akhileshthite) [inline src/server/apsystem.test.ts:257]: Ensure there's a check that the original activity is actually removed or marked as undone in some way. Currently, the test only checks if the undo activity is stored.

**Review by akhileshthite — COMMENTED** [review]
Thanks!
Few changes :)

**Comment by sutty-coop** [comment]
@RangerMauve I asked Agus about this and they said this is done and awaiting for your ok. 
Let us know how you see it. If it's done, I have to close this PR as well as issue 150 https://github.com/hyphacoop/distributed-press-organizing/issues/150

**Comment by akhileshthite** [comment]
> @RangerMauve I asked Agus about this and they said this is done and awaiting for your ok. 
> 
> Let us know how you see it. If it's done, I have to close this PR as well as issue 150 https://github.com/hyphacoop/distributed-press-organizing/issues/150

Hey, don't close this PR, it's not merged yet. I've requested some changes.

Ping @magush27

File: src/server/apsystem.test.ts (hunk 2 above)
Comment (magush27) [reply src/server/apsystem.test.ts:255]: Great, thanks! I accepted the changes and now I'm working on the updates and on this

**Comment by RangerMauve** [comment]
@magush27 the tests aren't passing, mind fixing them? Seems you're missing a function

**Comment by magush27** [comment]
> @magush27 the tests aren't passing, mind fixing them? Seems you're missing a function

yes, I'm with this right now

**Comment by magush27** [comment]
@RangerMauve the tests works well on push and in my local computer but not on pull_request, what is the problem?

**Comment by akhileshthite** [comment]
> @RangerMauve the tests works well on push and in my local computer but not on pull_request, what is the problem?

Hey, update your node version to `19.9.0` to resolve those gh action errors :)

**Comment by RangerMauve** [comment]
Ah, yeah merge the latest `main` branch into this.

`git pull origin/main` should do the trick

**Comment by fauno** [comment]
@RangerMauve @akhileshthite i merged main!

**Pull request merged**